File: frontend/src/features/GameCreate.js
import axios from 'axios';

export default async function GameCreateAPI(newGameData) {
    try {
        console.log(newGameData); // Log the input data

        const response = await axios.post('/api/game_create', newGameData, {
            headers: {
                'Content-Type': 'application/json', // Set the content type to JSON
            },
        });

        if (response.status === 200) {
            const createdGame = response.data;
            return createdGame; // Return the created team data
        } else {
            throw new Error(`Failed to create team: ${response.status}`);
        }
    } catch (error) {
        console.error('Game creation error:', error);
        throw error;
    }
}

File: backend/game_create.py
from bottle import post, request, response
import sqlite3
import json

# Create a new team
@post("/api/game_create")
def _():
    try:
        # Get the new team data from the request JSON
        new_game_data = request.json
        print("##### new game")
        print(new_game_data)  # Log the input data

        db = sqlite3.connect("./database/signifly-foosball.db", check_same_thread=False)
        print("##### connection succcesssful")


        # Insert the new team data into the database
        db.execute(
            """
            INSERT INTO games (team_id_1, team_id_2)
            VALUES (?, ?)
            """,
            (new_game_data["team_id_1"], new_game_data["team_id_2"])
        )

        db.commit()

        # Return a success response
        return {"message": "Game created successfully"}

    except Exception as ex:
        # Handle exceptions and return an error response if needed
        response.status = 500
        return {"error": str(ex)}
    finally:
        db.close()
